### PR TITLE
Fix for checkForFunctionProperty so that order does not matter

### PR DESCRIPTION
**Pull request by cwperks**
### Description

Fixes an issue with the implementation of a utility function used in vis_type_vega plugin where order matters when determining if an object has a property that is a function. This PR adds test to assert the desired behavior in multiple scenarios. 

### Check List

- [x] All tests pass
  - [x] `yarn test:jest`
  - [x] `yarn test:jest_integration`
- [x] New functionality includes testing.
- [x] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/vis_type_vega/public/data_model/utils.test.js [hunk 1]
```diff
@@ -91,4 +91,36 @@ describe('Utils.handleInvalidQuery', () => {
...
+        alert('Hello!');
+      },
+    };
+
+    expect(Utils.checkForFunctionProperty(input1)).toBe(true);
+
+    const input2 = {
+      key2: () => {
+        alert('Hello!');
+      },
```
Comment (seraphjiang) [inline src/plugins/vis_type_vega/public/data_model/utils.test.js:108]: could we use other method other than alert() in unit test. looks weird UT code has test dependencies on UI function.

just we just be?
```suggestion
      key2: () => {},
```

Comment (cwperks) [reply src/plugins/vis_type_vega/public/data_model/utils.test.js]: Replaced with jest.fn() which is much better. I was copying the same pattern as other tests in this fixture.

File: src/plugins/vis_type_vega/public/data_model/utils.test.js [hunk 2]
```diff
@@ -91,4 +87,30 @@ describe('Utils.handleInvalidQuery', () => {
       }).toThrowError();
     });
   });
+
+  test('should identify object contains function properties', async () => {
```
Comment (tianleh) [inline src/plugins/vis_type_vega/public/data_model/utils.test.js:91]: Good catch! For test completeness, could we have a test case whose result is `false`?

Comment (cwperks) [reply src/plugins/vis_type_vega/public/data_model/utils.test.js:91]: Added additional tests to cover negative cases.

**Review by tianleh — APPROVED** [review]

**Comment by cwperks** [comment]
@ashwin-pc @joshuarrrr Could I get some reviews on this PR?

**Review by SuZhou-Joe — APPROVED** [review]
LGTM.

**Review by bandinib-amzn — APPROVED** [review]
LGTM!

File: src/plugins/vis_type_vega/public/data_model/utils.ts [hunk 3]
```diff
@@ -79,14 +79,14 @@ export class Utils {
...
-    });
-    return result;
+    for (const value of Object.values(object)) {
+      if (typeof value === 'function') {
+        return true;
+      } else if (Utils.isObject(value) && Utils.checkForFunctionProperty(value)) {
+        return true;
+      }
+    }
+    return false;
```
Comment (seraphjiang) [inline src/plugins/vis_type_vega/public/data_model/utils.ts:89]: @cwperks  could we refactory code a little

```

    for (const value of Object.values(object)) {
      if (typeof value === 'function') return true;
      if (Utils.isObject(value) && Utils.checkForFunctionProperty(value)) return true;
    }
    return false;
```

**Pull request merged**